craft_providers/bases/checks.py

```diff
@@ -0,0 +1,138 @@
+#
+# Copyright 2021-2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""Base compatibility checks."""
+
+import enum
+import io
+import logging
+import pathlib
+import platform
+import subprocess
+from functools import total_ordering
+from textwrap import dedent
+from typing import Dict, List, Optional, cast
+
+from craft_providers.actions.snap_installer import Snap
+from craft_providers.bases.ubuntu import BuilddBase, BuilddBaseAlias
+from craft_providers.base import Base
+from craft_providers.errors import (
+    BaseCompatibilityError,
+    BaseConfigurationError,
+    ProviderError,
+    details_from_called_process_error,
+)
+from craft_providers.executor import Executor
+from craft_providers.util.os_release import parse_os_release
+
+logger = logging.getLogger(__name__)
+
+
+"""
+    XENIAL = "16.04"
+    BIONIC = "18.04"
+    FOCAL = "20.04"
+    JAMMY = "22.04"
+    NOBLE = "24.04"
+    ORACULAR = "24.10"
+    DEVEL = "devel"
+"""
+
+
+INVALID_VERSIONS = [
+    {
+        "host_less_than_equal": BuilddBaseAlias.FOCAL,
+        "guest_greater_than_equal": BuilddBaseAlias.ORACULAR,
+        # The system is affected by the cgroups bug if both of the above and either of the below
+        "lxd_less_than": [
+            (5, 0, 4),
+            (5, 21, 2),
+        ],
+        "kernel_less_than": (5, 15),
```

Comment:
How about making this a list of triples as well and using the same function for matching lxd and kernels?

Reply:
It's not necessary now, but it could be necessary in the future, especially given OEM kernels.